spring-integration-groovy/src/test/java/org/springframework/integration/groovy/ScriptsFactoryTests.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.groovy;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.handler.MessageProcessor;
+import org.springframework.integration.scripting.dsl.Scripts;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.integration.test.util.TestUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * @author Artem Bilan
+ * @since 5.0
+ */
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class ScriptsFactoryTests {
+
+	@Autowired
+	@Qualifier("scriptFilter.input")
+	private MessageChannel filterInput;
+
+	@Autowired
+	private PollableChannel discardChannel;
+
+	@Autowired
+	private PollableChannel results;
+
+	@Autowired
+	private MessageProcessor<?> scriptMessageProcessor;
+
+	@Test
+	public void filterTest() {
```

Comment:
The main purpose of this test to be sure that we really use `GroovyScriptExecutingMessageProcessor` underneath.
Everything rest is covered by other existing tests.
We don't need to worry here about `.route()`, `.handle()` etc., because our main goal to test the proper `MessageProcessor`. 
The same one is used everywhere.
Just take a look to the syntax:
```
handle(MessageProcessorSpec<?> messageProcessorSpec)
transform(MessageProcessorSpec<?> messageProcessorSpec)
split(MessageProcessorSpec<?> messageProcessorSpec)
```
and so on.